doc/porting_guide.md

```diff
@@ -3,268 +3,284 @@
 
 ## Table of Contents
 1. [Target Audience](#target-audience)
-2. [Xively Client versions](#xively-client-versions)
+2. [C Client Overview](#c-client-overview)
 3. [Supported Platforms](#supported-platforms)
 4. [Board Support Package (BSP)](#board-support-package-bsp)
 5. [Porting the C Client](#porting-the-c-client)
-    - [Preceding Step - A Native Build](#preceding-step---a-native-build)
-    - [The Porting Step](#the-porting-step)
+    - [Suggested First Step - A Native Build](#suggested-first-step---a-native-build)
+    - [Porting the Xively C Client to Your Platform](#porting-the-xively-c-client-to-your-platform)
 6. [Xively Client Features](#xively-client-features)
 
 
 ## Target Audience
-This document is intended for Developers who have access to the source code of the Xively C Client. Developers who want to port the Xively C to a particular platform will find this to be the main guide for that endeavor.
+This document is intended for Embedded Developers who have access to the source code of the Xively C Client and want to run it on their custom platform.
 
-Details of build, config flags and the location of platform-specific code are included in this document.
+Details of the Xively C Client build steps, build configuration flags, and the location of the functions needed to port the library to new platforms are included in this document.
 
-## Xively Client versions
+For information about the Xively Python client, please visit its public git repository: [xiPy](https://github.com/xively/xipy).
 
-#### C Client
+## C Client Overview
 
-The C Client is designed for low resources platforms. The single threaded implementation is non-blocking, which allows it to operate on NoOs platforms. It also unlocks the ability for processing outgoing and incoming messages concurrently.
+The Xively C Client is designed for low resources platforms. The single threaded implementation is non-blocking, which allows it to operate on NoOs platforms. The implementation also supports seamless concurrent processing of outgoing and incoming messages.
 
-An event system is included and can be used by client applications to schedule publications at regular intervals without any platform-specific timers or tasks.
+An event system is included in the client which can be used by your client application as well. The client's event system is a platform independant implementation that doesn't require porting to any platform-specific timers or tasks.
 
 The client provides this functionality while remaining small enough to reside on low resource MCUs, consumes minimal memory, processor power and energy consumption.
```

Comment:
I know you haven't touched this sentence but "consume" appears twice. Could you rephrase it?